### PR TITLE
Core/Unit: remove emote state when creature attacks another unit

### PR DESCRIPTION
**Pull request by SnapperRy**
**Changes proposed**:

Prevents visual bugs where the creature keeps dancing/sitting/kneeling when in combat. Examples: guids 126034 and 59439 (thanks Killyana).

**Target branch(es)**: 335

**Tests performed**: tested and working..

### CODE REVIEW

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -8584,6 +8584,9 @@ bool Unit::Attack(Unit* victim, bool meleeAttack)
 
         ToCreature()->SendAIReaction(AI_REACTION_HOSTILE);
         ToCreature()->CallAssistance();
+
+        // Remove emote state - will be restored on creature reset
+        SetUInt32Value(UNIT_NPC_EMOTESTATE, 0);
```
Comment (Keader) [inline src/server/game/Entities/Unit/Unit.cpp]: SetUInt32Value(UNIT_NPC_EMOTESTATE, EMOTE_ONESHOT_NONE);

Comment (SnapperRy) [inline src/server/game/Entities/Unit/Unit.cpp]: Changed, thanks.

**Comment by tkrokli** [comment]
Thanks for looking into that funny bug. I haven't seen it myself, but I try to imagine /dance going on in a boss fight or something similar :laughing:

**Comment by Treeston** [comment]
No reliable repro, but I've seen this makes NPCs stutter while attacking sometimes.

Maybe only reset the emote state if it isn't already `NONE`?

**Pull request merged**